### PR TITLE
Add GitHub Actions to do our CI Builds. Will build in parallel and many more jobs than now.

### PR DESCRIPTION
**Pull request by HHHartmann**
Fixes #No Issue.

- [x] This PR is for the `dev` branch rather than for the `release` branch.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [x] I have thoroughly tested my contribution.
- [x] The code changes are reflected in the documentation at `docs/*`.

\<Description of and rationale behind this PR\>
This adds An Action to do many checks on commits on PRs.
Many tests are added new compared to the travis build.

Currently this PR does not execute the Actions for some reason. See https://github.com/HHHartmann/nodemcu-firmware/pull/1 for a working example.

At the moment there are following steps:

CI / build (51, float) (pull_request) Successful in 1m
CI / build (51, integral) (pull_request) Successful in 1m
CI / build (53, float) (pull_request) Successful in 1m
CI / build_luac_cross_win (pull_request) Successful in 1m — build_luac_cross_win
CI / luacheck (linux, ubuntu-latest) (pull_request) Successful in 26s
CI / luacheck (windows, windows-latest) (pull_request) Successful in 20s
CI / compile_lua (51, float, cat) (pull_request) Successful in 7s
CI / compile_lua (51, integral, grep -v "lua_modules/lm92/lm92.lua\|lua_modules/hdc1000/HDC1000.lua\|l... (pull_request) Successful in 6s
CI / compile_lua (53, float, cat) (pull_request) Successful in 7s
CI / NTest (51, float) (pull_request) Successful in 6s
CI / NTest (51, integral) (pull_request) Successful in 6s
CI / NTest (53, float) (pull_request) Successful in 6s

More tests for win luac.cross to come as soon as https://github.com/nodemcu/nodemcu-firmware/pull/3336 is merged

### CODE REVIEW

**Comment by pjsg** [comment]
Can you add "build (53, 64BIT) "? It just needs -DLUA_NUMBER_64BITS on the make line.

**Comment by HHHartmann** [comment]
> Can you add "build (53, 64BIT) "? It just needs -DLUA_NUMBER_64BITS on the make line.

Works!

**Comment by marcelstoer** [comment]
Great, this is coming along nicely!

**Comment by HHHartmann** [comment]
Sure would make things easier having the separate report for each test with this PR

**Comment by HHHartmann** [comment]
Is there any chance to get this merged soon?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,242 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
```
Comment (marcelstoer) [inline .github/workflows/build.yml:5]:
```suggestion
    branches: [ release ]
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+      fail-fast: false
+      matrix:
+        lua_ver: [51, 53]
+        numbers: ['float']
+        include:
+          - lua_ver: 51
+            numbers: 'integral'  
+          - lua_ver: 53
+            numbers: '64bit'  
+    runs-on: ubuntu-latest
```
Comment (marcelstoer) [inline .github/workflows/build.yml:23]: 1. "latest" here seems to imply v18.04. On Travis, however, we use Xenial/16.04 as [we don't specify `dist: something`](https://github.com/nodemcu/nodemcu-firmware/blob/release/.travis.yml) and [Xenial is the default.](https://docs.travis-ci.com/user/reference/overview/). Is the 16->18 update intentional?
2. IMO we should pin the version rather then rely on whatever GitHub uses as "latest".

File: .github/workflows/build.yml (hunk 1 above)
Comment (HHHartmann) [reply .github/workflows/build.yml:5]: Did I really write this ?

File: .github/workflows/build.yml (hunk 2 above)
Comment (HHHartmann) [reply .github/workflows/build.yml:23]: Pinned. In docker we also use Xenial.

**Pull request merged**